### ISSUE TITLE
Segfault when objects have custom `__mpz__` method raising errors

### ISSUE THREAD

**Issue by videlec** [aleaxit/gmpy#173]
```
>>> import gmpy2
>>> class A:
...    def __mpz__(self): raise TypeError
>>> a = A()
>>> gmpy2.mpz(a)
** SEGFAULT **
```

**Comment by vinklein** [comment]
Same problem with __mpq__, __mpfr__ and __mpc__ function.
All fixed in PR #174

**Comment by casevh** [comment]
Thanks.
